mne/io/snirf/tests/test_snirf.py

```diff
@@ -413,25 +443,80 @@ def test_snirf_fieldtrip_od():
 
 
 @requires_testing_data
-def test_snirf_kernel_hb():
-    """Test reading Kernel SNIRF files with haemoglobin data."""
-    raw = read_raw_snirf(kernel_hb, preload=True)
-
-    # Test data import
-    assert raw._data.shape == (180 * 2, 14)
-    assert raw.copy().pick("hbo")._data.shape == (180, 14)
-    assert raw.copy().pick("hbr")._data.shape == (180, 14)
-
-    assert_allclose(raw.info["sfreq"], 8.257638)
+@pytest.mark.parametrize(
+    "kind, ver, shape, n_nan, fname",
+    [
+        pytest.param("hb", "new", (4, 38), 0, kernel_hb, id="hb"),
+        pytest.param("hb", "old", (180 * 2, 14), 20, kernel_hb_old, id="hb old"),
+        pytest.param(
+            "td moments", "new", (12, 38), 0, kernel_td_moments, id="td moments"
+        ),
+        pytest.param("td gated", "new", (100, 38), 0, kernel_td_gated, id="td gated"),
+        pytest.param(
+            "td moments",
+            "old",
+            (1080, 14),
+            60,
+            kernel_td_moments_old,
+            id="td moments old",
+        ),
+    ],
+)
+def test_snirf_kernel_basic(kind, ver, shape, n_nan, fname):
+    """Test reading Kernel SNIRF files with haemoglobin or TD data."""
+    raw = read_raw_snirf(fname, preload=True)
+    if kind == "hb":
+        # Test data import
+        assert raw._data.shape == shape
+        hbo_data = raw.get_data("hbo")
+        hbr_data = raw.get_data("hbr")
+        assert hbo_data.shape == hbr_data.shape == (shape[0] // 2, shape[1])
+        hbo_norm = np.nanmedian(np.linalg.norm(hbo_data, axis=-1))
+        hbr_norm = np.nanmedian(np.linalg.norm(hbr_data, axis=-1))
+        # TODO: Old file vs new file scaling, one is wrong!
+        if ver == "new":
+            assert 1e-5 < hbr_norm < hbo_norm < 1e-4
+        else:
+            assert 1 < hbr_norm < 3
+    elif kind == "td moments":
+        assert raw._data.shape == shape
+        n_ch = 0
+        # TODO: Reasonable values here???
+        lims = dict(intensity=(1e4, 1e7), mean=(1e3, 1e4), variance=(1e5, 1e7))
```

Comment:
I also don't know if these are reasonable values for the counts (intensity), mean (s), or variance (s**2)

Reply:
these are reasonable if mean is in picoseconds and variance in ps^2.